src/aatoolbox/utils/raster.py

```diff
@@ -562,7 +588,42 @@ def compute_raster_stats(
         -------
         List[pandas.DataFrame]
             List of raster statistics data frames.
+
+        Examples
+        --------
+        >>> import geopandas as gpd
+        >>> import xarray as xr
+        >>> import rioxarray
+        >>> from shapely.geometry import Polygon
+        >>>
+        >>> # compute raster stats on simple data
+        >>> d = {
+        ...     "name": ["area_a", "area_b"],
+        ...     "geometry": [
+        ...         Polygon([(0, 0), (0, 2), (2, 2), (2, 0)]),
+        ...         Polygon([(2, 0), (2, 2), (3, 2), (3, 0)]),
+        ...     ],
+        ... }
+        >>> gdf = gpd.GeoDataFrame(d)
+        >>>
+        >>> ds = xr.DataArray(
+        ...     [[1, 2, 3], [4, 5, 6]],
+        ...     dims=("y", "x"),
+        ...     coords={"y": [1.5, 0.5], "x": [0.5, 1.5, 2.5]},
+        ... ).rio.write_crs("EPSG:4326").to_dataset(name="data")
+        >>>
+        >>> ds.aat.compute_raster_stats(["data"], gdf=gdf, feature_col="name")
         """
-        vars = self.vars if vars is None else vars
-        stats = [self._obj[var].compute_raster_stats(**kwargs) for var in vars]
+        if vars is None:
+            vars = self.vars
+        else:
+            if not isinstance(vars, list):
```

Comment:
Ha, yeah, again, I think coming from R the default is to enforce explicit typing to prevent any unexpected behavior, as methods and functions there work in subtly different ways on different classes. Will switch to your method.